common/network_manager.go

```diff
@@ -588,7 +607,7 @@ func (m *NetworkManager) updateProtocolCacheDisabled() error {
 }
 
 func (m *NetworkManager) updateProtocolRequestInterception() error {
-	enabled := m.userReqInterceptionEnabled
+	enabled := m.userReqInterceptionEnabled || m.credentials != nil
```

Comment:
Checking `m.credentials` here feels out of place. I think we should do this earlier, [where we set `m.userReqInterceptionEnabled`](https://github.com/grafana/xk6-browser/blob/81bac670602c83836042f7fdba0a6740f8f8460b/common/network_manager.go#L578) maybe?

Reply:
I don't think we need to follow Playwright code line-for-line, even less so with trivial things like this.

I mention it because if `m.credentials == nil` we probably want to have `m.userReqInterceptionEnabled == false`, so it makes more sense to set it beforehand, to avoid cluttering up this code. I considered doing this even higher up, in [`FrameSession.updateRequestInterception()`](https://github.com/grafana/xk6-browser/blob/81bac670602c83836042f7fdba0a6740f8f8460b/common/frame_session.go#L999), where we already do the `hasRoutes()` check, but that would involve checking an unexported `FrameManager` field, which might become a problem later on if we refactor this, so in `setRequestInterception` is probably fine.

Reply:
I think a better place for setting `userReqInterceptionEnabled` can be `NetworkManager.Authenticate`. I'll try this next week and see how it goes.

https://github.com/grafana/xk6-browser/blob/81bac670602c83836042f7fdba0a6740f8f8460b/common/network_manager.go#L624-L631

> I don't think we need to follow Playwright code line-for-line, even less so with trivial things like this.

Robin followed it almost line by line, and until we stabilize the codebase, it can tremendously help us add new features and fix bugs. So I can't say I agree :'(

Reply:
I think a better place for setting `userReqInterceptionEnabled` may be `NetworkManager.Authenticate`. I'll try this and see how it goes.

https://github.com/grafana/xk6-browser/blob/81bac670602c83836042f7fdba0a6740f8f8460b/common/network_manager.go#L624-L631

> I don't think we need to follow Playwright code line-for-line, even less so with trivial things like this.

It's helpful until we stabilize the codebase ;)